File: Source/Alternet.UI.Pal/AuiManager.cpp
#include "AuiManager.h"

namespace Alternet::UI
{
	wxAuiManager* AuiManager::Manager(void* handle)
	{
		return (wxAuiManager*)handle;
	}

	void AuiManager::Delete(void* attr) 
	{
		delete (wxAuiManager*)attr;
	}

	void* AuiManager::CreateAuiManager()
	{
		return new wxAuiManager();
	}

	AuiManager::AuiManager()
	{

	}

	AuiManager::~AuiManager()
	{
	}

	void* AuiManager::CreateAuiManager2(void* managedWnd, uint32_t flags)
	{
		return new wxAuiManager((wxWindow*)managedWnd, flags);
	}

	void AuiManager::UnInit(void* handle)
	{
	}

	void AuiManager::SetFlags(void* handle, uint32_t flags)
	{
	}

	uint32_t AuiManager::GetFlags(void* handle)
	{
		return 0;
	}

	bool AuiManager::AlwaysUsesLiveResize()
	{
		return false;
	}

	bool AuiManager::HasLiveResize(void* handle)
	{
		return false;
	}

	void AuiManager::SetManagedWindow(void* handle, void* managedWnd)
	{
	}

	void* AuiManager::GetManagedWindow(void* handle)
	{
		return nullptr;
	}

	void* AuiManager::GetManager(void* window)
	{
		return nullptr;
	}

	void AuiManager::SetArtProvider(void* handle, void* artProvider)
	{
	}

	void* AuiManager::GetArtProvider(void* handle)
	{
		return nullptr;
	}

	bool AuiManager::DetachPane(void* handle, void* window)
	{
		return false;
	}

	void AuiManager::Update(void* handle)
	{
	}

	string AuiManager::SavePerspective(void* handle)
	{
		return wxStr(wxEmptyString);
	}

	bool AuiManager::LoadPerspective(void* handle, const string& perspective, bool update)
	{
		return false;
	}

	void AuiManager::SetDockSizeConstraint(void* handle, double widthPct,
		double heightPct)
	{
	}

	void AuiManager::RestoreMaximizedPane(void* handle)
	{
	}

	void* AuiManager::GetPane(void* handle, void* window)
	{
		return nullptr;
	}

	void* AuiManager::GetPaneByName(void* handle, const string& name)
	{
		return nullptr;
	}

	bool AuiManager::AddPane(void* handle, void* window, void* paneInfo)
	{
		return false;
	}

	bool AuiManager::AddPane2(void* handle, void* window, void* paneInfo,
		double dropPosX, double dropPosY)
	{
		return false;
	}

	bool AuiManager::AddPane3(void* handle, void* window, int direction,
		const string& caption)
	{
		return false;
	}

	bool AuiManager::InsertPane(void* handle, void* window, void* insertLocPaneInfo,
		int insertLevel)
	{
		return false;
	}

	string AuiManager::SavePaneInfo(void* handle, void* paneInfo)
	{
		return wxStr(wxEmptyString);
	}

	void AuiManager::LoadPaneInfo(void* handle, const string& panePart, void* paneInfo)
	{
	}

	Size AuiManager::GetDockSizeConstraint(void* handle)
	{
		return Size();
	}

	void AuiManager::ClosePane(void* handle, void* paneInfo)
	{
	}

	void AuiManager::MaximizePane(void* handle, void* paneInfo)
	{
	}

	void AuiManager::RestorePane(void* handle, void* paneInfo)
	{
	}

	void* AuiManager::CreateFloatingFrame(void* handle, void* parentWindow,
		void* paneInfo)
	{
		return nullptr;
	}

	bool AuiManager::CanDockPanel(void* handle, void* paneInfo)
	{
		return false;
	}
}

File: Source/NativeApi/NativeApi/Api/AuiManager/AuiManager.cs
#pragma warning disable
using ApiCommon;
using System;
using System.Collections.Generic;
using System.Text;
using Alternet.Drawing;

namespace NativeApi.Api
{
	//https://docs.wxwidgets.org/3.2/classwx_aui_manager.html
	public class AuiManager
	{
        public static void Delete(IntPtr handle) { }

        public static IntPtr CreateAuiManager() => throw new Exception();
        public static IntPtr CreateAuiManager2(IntPtr managedWnd,
            uint flags) => throw new Exception();

        public static void UnInit(IntPtr handle) {}

        public static void SetFlags(IntPtr handle, uint flags) { }
        public static uint GetFlags(IntPtr handle) => throw new Exception();

        public static bool AlwaysUsesLiveResize() => throw new Exception();
        public static bool HasLiveResize(IntPtr handle) => throw new Exception();

        public static void SetManagedWindow(IntPtr handle, IntPtr managedWnd) { }
        public static IntPtr GetManagedWindow(IntPtr handle) => throw new Exception();

        public static IntPtr GetManager(IntPtr window) => throw new Exception();

        public static void SetArtProvider(IntPtr handle, IntPtr artProvider) {}
        public static IntPtr GetArtProvider(IntPtr handle) => throw new Exception();

        public static bool DetachPane(IntPtr handle, IntPtr window) =>
            throw new Exception();

        public static void Update(IntPtr handle) {}

        public static string SavePerspective(IntPtr handle) => throw new Exception();
        public static bool LoadPerspective(IntPtr handle,
            string perspective, bool update) => throw new Exception();

        public static void SetDockSizeConstraint(IntPtr handle, double widthPct,
            double heightPct) => throw new Exception();
        public static void RestoreMaximizedPane(IntPtr handle) =>
            throw new Exception();

        public static IntPtr GetPane(IntPtr handle, IntPtr window) =>
            throw new Exception();
        public static IntPtr GetPaneByName(IntPtr handle, string name) =>
            throw new Exception();
        //public static IntPtr[] GetAllPanes(IntPtr handle) => throw new Exception();

        public static bool AddPane(IntPtr handle, IntPtr window, IntPtr paneInfo) =>
            throw new Exception();

        public static bool AddPane2(IntPtr handle, IntPtr window, IntPtr paneInfo,
            double dropPosX,
            double dropPosY) => throw new Exception();

        public static bool AddPane3(IntPtr handle, IntPtr window, int direction,
            string caption) => throw new Exception();

        public static bool InsertPane(IntPtr handle, IntPtr window,
            IntPtr insertLocPaneInfo,
                     int insertLevel) => throw new Exception();

        public static string SavePaneInfo(IntPtr handle, IntPtr paneInfo) =>
            throw new Exception();
        public static void LoadPaneInfo(IntPtr handle, string panePart,
            IntPtr paneInfo) => throw new Exception();

        public static SizeD GetDockSizeConstraint(IntPtr handle) => throw new Exception();

        public static void ClosePane(IntPtr handle, IntPtr paneInfo) =>
            throw new Exception();
        public static void MaximizePane(IntPtr handle, IntPtr paneInfo) =>
            throw new Exception();
        public static void RestorePane(IntPtr handle, IntPtr paneInfo) =>
            throw new Exception();

        public static IntPtr CreateFloatingFrame(IntPtr handle,
            IntPtr parentWindow, IntPtr paneInfo) => throw new Exception();
        public static bool CanDockPanel(IntPtr handle, IntPtr paneInfo) =>
            throw new Exception();
    }
}